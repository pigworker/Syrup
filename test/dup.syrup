dup(<Bit>) -> <Bit>, <Bit>
dup(X) = X,X

experiment dup

not(<Bit>) -> <Bit>
not(X) = nand(X,X)

dop(<Bit>) -> <Bit>, <Bit>
dop(X) = dup(!X)

experiment dop

doop(<Bit>) -> <Bit>, <Bit>, <Bit>, <Bit>
doop(X) = dup(!X), dup(!X)

experiment doop
