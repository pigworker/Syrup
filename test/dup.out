Circuit `dup` is defined.

Truth table for `dup`:
  X | X X
  --|----
  0 | 0 0
  1 | 1 1

Circuits `not` and `dop` are defined.

Truth table for `dop`:
  X |    
  --|----
  0 | 1 1
  1 | 0 0

Circuit `doop` is defined.

Truth table for `doop`:
  X |        
  --|--------
  0 | 1 1 1 1
  1 | 0 0 0 0
